stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/extensions/ChatClient.kt

```diff
@@ -362,23 +363,17 @@ private suspend fun ChatClient.loadMessageByIdInternal(
         return cidValidationResult.error().toResultError()
     }
 
-    val result = getMessage(messageId).await()
+    val (channelType, channelId) = cid.cidToTypeAndId()
+    val result = logic.channel(channelType = channelType, channelId = channelId)
+        .loadMessagesAroundId(messageId)
 
     return if (result.isSuccess) {
-        val message = result.data()
-        val (channelType, channelId) = cid.cidToTypeAndId()
-
-        logic.channel(channelType = channelType, channelId = channelId).run {
-            storeMessageLocally(listOf(message))
-            loadMessagesAroundId(messageId)
+        result.map { channel ->
+            channel.messages.first { message ->
```

Comment:
This might throw `NoSuchElementException` if the message is not there. What do you think about replacing it with `firstOrNull` and returning `Result.error` just in case the message is not there?